### PR TITLE
Add a few modules.go tests and slightly improve caching code

### PR DESCRIPTION
**Pull request by wenovus**
If we don't cache negative FindModule* results (which we shouldn't because `modules.Read` could add new ones), then the first if block can never happen, so removing those.

### CODE REVIEW

File: pkg/yang/modules_test.go [hunk 1]
```diff
@@ -130,6 +132,247 @@ func TestModulesFindByNamespace(t *testing.T) {
...
+				module sysdb {
+					prefix s;
+					namespace "urn:s";
+
+					revision 01-01-01 { description "the start of time"; }
+
+					container sys { leaf hostname { type string; } }
+				}`,
+		},
+		wantErrSubstr: "no such module",
```
Comment (robshakir) [inline pkg/yang/modules_test.go:165]: Perhaps remove the second module here -- it took me a moment to spot that `sys` != `sysdb` in the imports.

**Review by robshakir — APPROVED** [review]

File: pkg/yang/modules.go [hunk 2]
```diff
@@ -219,20 +218,20 @@ func (ms *Modules) FindModuleByNamespace(ns string) (*Module, error) {
...
 		return nil, fmt.Errorf("%s: no such namespace", ns)
 	}
+	// Don't cache negative results because new modules could be added.
+	ms.byNS[ns] = found
 	return found, nil
 }
 
 // FindModuleByPrefix either returns the Module specified by prefix or returns
 // an error.
+// TODO(wenovus): This should be deprecated since prefixes are not unique among
```
[the remark this answers is not in the document]
Comment (wenovus) [reply pkg/yang/modules.go:231]: Yes that's right.

File: pkg/yang/modules_test.go (hunk 1 above)
Comment (wenovus) [reply pkg/yang/modules_test.go:156]: Done.

File: pkg/yang/modules.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (wenovus) [reply pkg/yang/modules.go:231]: The only current user is a call from an Entry object, so perhaps we can add the entry as an input parameter.

**Pull request merged**